### DISCUSSION TITLE
Module import?

### DISCUSSION THREAD

**Discussion by MelvinKlein** [argoproj-labs/hera#749] [Q&A] [answered]
Hi, I'm new to Hera and have a question on how this is supposed to work.

All the examples only consist of one file that includes all the code. I tried to write very basic task/script in which I import a function from another module and execute it. This did not work at all.
I printed the created *.yaml and it did not include any of the imported code, just the import statement.
My guess is that this is not supported and the image in which the code gets executed has to provide all the imports?
If that is the case, is there a way build into hera to automate the build and upload of a docker-image that includes all of the code?

Here are is the code I used:
```
from hera.workflows import Steps, Workflow, script
from hera.shared import global_config

global_config.host = "https://127.0.0.1:2746"
global_config.verify_ssl = False



@script()
def test_it() -> None:
    from printer import do
    do("a")



with Workflow(
        generate_name="test-workflow-",
        namespace="argo",
        entrypoint="steps",
) as w:
    with Steps(name="steps"):
        test_it()

w.create()
print(w.to_yaml())
```

and the printer.py
```
def do(argument):
    print(argument)
```

**Comment by flaviuvadan** [comment, marked as answer]
Thanks for the question @MelvinKlein! 

> My guess is that this is not supported and the image in which the code gets executed has to provide all the imports?

This is correct. The submitted function executes in a container. That container env must have all the imports available

> is there a way build into hera to automate the build and upload of a docker-image that includes all of the code?

Not at the moment. Hera does not have a running daemon/agent, GitHub Action, etc, that could do that atm. Does your org/personal project have a CI setup that can construct a Docker image upon pushing new code? That's one approach to use - push the code, build the image, create the workflow (with remote caching and what not building an image may not take too long)

**Reply by MelvinKlein** [reply]
> This is correct. The submitted function executes in a container. That container env must have all the imports available

Thanks for the clarification.

> Not at the moment. Hera does not have a running daemon/agent, GitHub Action, etc, that could do that atm. Does your org/personal project have a CI setup that can construct a Docker image upon pushing new code? That's one approach to use - push the code, build the image, create the workflow (with remote caching and what not building an image may not take too long)

Yes we can do that.
